aten/src/THC/THCTensorMathReduce.cuh

```diff
@@ -503,12 +476,10 @@ THC_transformReduceInnermostDimIndex(THCState *state,
     num_rows *= THCTensor_sizeLegacyNoScalars(state, src, dim);
   }
   unsigned row_size = THCTensor_sizeLegacyNoScalars(state, src, ndim - 1);
-
-  dim3 threads(16, 32);
-  dim3 grid(min(1024, THCCeilDiv(num_rows, threads.y)));
+  int blocks = min(1024, num_rows);
```

Comment:
From @wesolwsk: can you try without this limit? 1024 blocks yields occupancy of about 13 on Volta for short rows. Maybe you can get better performance for large input sizes if you just set the number of blocks to the row count. It would also allow you to get rid of the external for loop, which could help a little more.